### PR TITLE
Deprecate/refactor content in docs/specification

### LINKED ISSUE #1814: deal with content in docs/specification

### PR DESCRIPTION
**Pull request by zramsay**
closes #1814 
replaces #1914 

* [it is docs] Updated all relevant documentation in docs
* [no code] Updated all code comments where relevant
* [x] Wrote tests
* [ ] Updated CHANGELOG.md

### CODE REVIEW

File: docs/tendermint-core/running-in-production.md [hunk 1]
```diff
@@ -104,6 +104,69 @@ signals we use the default behaviour in Go: [Default behavior of signals
...
+       then remove all the lines starting from the corrupted one and restart
+       Tendermint.
+
+```
+$EDITOR /tmp/corrupted_wal
+```
+  5. After editing, convert this file back into binary form by running:
+
+```
+./scripts/json2wal/json2wal /tmp/corrupted_wal > "$TMHOME/data/cs.wal/wal"
```
[the remark this answers is not in the document]
Comment (xla) [reply docs/tendermint-core/running-in-production.md]: @melekes Can you create a follow-up issue?

[the remark this answers is not in the document]
Comment (melekes) [reply docs/tendermint-core/running-in-production.md]: https://github.com/tendermint/tendermint/issues/1954

File: docs/specification/wire-protocol.rst [hunk 2]
```diff
@@ -1,172 +0,0 @@
-Wire Protocol
```
[the remark this answers is not in the document]
Comment (zramsay) [reply docs/specification/wire-protocol.rst]: https://github.com/tendermint/tendermint/pull/1914/files#diff-9d5f3395117a1a66036b0ac554aae7f0R3

[the remark this answers is not in the document]
Comment (zramsay) [reply docs/specification/wire-protocol.rst]: the docs landing page will need improvements after this PR, which will point people to various places

**Pull request merged**